### DISCUSSION TITLE
URP support?

### DISCUSSION THREAD

**Discussion by etiennepinchon** [icosa-gallery/open-brush#96] [Q&A] [answered]
Hi, 

Any plan to support URP rather than only Standard?

Thanks!

**Comment by TimAidley** [comment, marked as answer]
Hi Etienne!
  We did have a chat about this in the discord, and I think the conclusion was that we probably would want to convert over to URP at some point. I don't know of anybody who is working on it at the moment, though.

The thing about it as a task is that it's probably a fair bit of work, but if done properly will result in an Open Brush that looks and acts exactly the same as the version that uses the legacy renderer. I doubt anyone will do it until they have some other feature that they want to do that would require them to switch everything to URP first.

**Reply by mikeskydev** [reply]
I believe @andybak was looking at the possibility of adding URP shaders to the tilt brush toolkit so that other developers could easily import correct looking sketches to their URP projects, I guess this would go hand in hand.

**Comment by andybak** [comment]
@etiennepinchon - I am indeed intending to look at this.

Can you clarify what you mean by URP support at this stage? 

Are you looking for a build of Open Brush that runs on URP? If so - what do you want to achieve?

Are there URP specific features you need?

Or is this more about supporting export and rendering of sketches in Unity for other projects?

It's the latter that we will probably tackly first. But if it's the former that interests you then info on what your use-cases are would be helpful.

**Reply by tacopancakes** [reply]
URP being the Universal Render pipeline. I would think its the ability to use the Shader Graph visual coding tools. It solves many headaches with making custom render passes and lends itself to artists. BUT! BIG BUT, custom shaders are a super pain right now for GLTF, one they are not available by default, and two the web viewer being THREE.JS requires a post processor either in their hook system or the never yet documented Nodes material system. The Nodes system could have a simple convertor at some point. But that would require collaboration.

**Reply by andybak** [reply]
I'm still trying to clarify what exactly the this issue is about. "Support URP" could mean several things so it would be handy if @etiennepinchon could explain their workflow. (and anyone else of course).

I suspect this is about Tilt Brush Toolkit - in which case we should really move the issue to https://github.com/icosa-gallery/open-brush-toolkit/issues

**Reply by mikeskydev** [reply]
I would also assume this is more suited to TBT, in theory it shouldn't matter what Open Brush uses internally, just that it's easy to display creations in other unity apps.

**Comment by andybak** [comment]
Someone seems to be working on it: https://github.com/kijai/tilt-brush-toolkit/releases/tag/v23.0.1-URP-alpha_v02

@kijai - come and talk to us!

**Reply by kijai** [reply]
Hey, yeah I've been working on a fork of the tiltbrush-toolkit to convert/remake the shaders that aren't URP compatible. This effort has been purely to allow exports to work in AltspaceVR -platform, which is switching over the URP next month, thus it includes some workarounds and extra options in some of the shaders since the platform in question do not support post processing.

I'm no expert in shaders and most of these changes are simply shadergraphs, but all the brushes do work with URP. I've tested this with both TiltBrush .fbx exports and Open Brush .glb exports.

**Reply by andybak** [reply]
Hey @kijai - thanks for popping up and apologies for the slow response. I'm stuck on other stuff at the moment but URP support is a common request and your work is undoubtably going to prove useful even if we need to tweak it. I've posted a link to your repo in the Discord in the thread where people where discussing this.
